### ISSUE TITLE
flux-kontext train: Latent Pre-caching and Text Encoder Output Pre-caching bug

### ISSUE THREAD

**Issue by xyxxmb** [kohya-ss/musubi-tuner#400]
I found that during flux-kontext training, when using jsonl data format to construct triplets <image_path, control_path, caption>, if there are two entries with different control_paths and captions but the same image_path, then during the Latent Pre-caching phase, since cache_latent is named after the image_path, there will only be one entry in the cache_directory (the latter one replaces the former).

Similarly, during the Text Encoder Output Pre-caching phase when extracting text features from captions, a warning appears: WARNING:musubi_tuner.dataset.image_video_dataset:caption mismatch: existing=captionA, new=captionB, overwrite.

This means that if multiple triplets share the same image_path, it will lead to data loss and risk of incorrect Text Encoder Output Pre-caching extraction.

**Comment by kohya-ss** [comment]
Currently, one image can only have one control image. Please copy and rename the image to pair it with each control image.

**Comment by xyxxmb** [comment]
> Currently, one image can only have one control image. Please copy and rename the image to pair it with each control image.

Thanks，Are there any plans to support this feature later?

**Comment by icekeg** [comment]
My coml file content is as follows：

[general]
resolution = [1024, 1024]
caption_extension = ".txt"
batch_size = 1
num_repeats = 1
enable_bucket = true
bucket_no_upscale = false

[[datasets]]
flux_kontext_no_resize_control = true
image_directory = "/workspace/team_code_faidrawing/comfy_mtalk/train/kontext/blocksketch/traget"
control_directory = "/workspace/team_code_faidrawing/comfy_mtalk/train/kontext/blocksketch/crl"
cache_directory = "/workspace/team_code_faidrawing/comfy_mtalk/train/kontext/blocksketch/cache"


during the Latent Pre-caching phase, only one cache file is generated for the control picture and image picture. Is this correct?

**Comment by icekeg** [comment]
The loss value during training is always nan.

<img width="1267" height="75" alt="Image" src="https://github.com/user-attachments/assets/b19d3911-d0c0-4feb-b160-70567ea770f6" />

**Comment by kohya-ss** [comment]
> during the Latent Pre-caching phase, only one cache file is generated for the control picture and image picture. Is this correct?

If you have multiple images, make sure the filenames are correct, you must follow the rules in the documentation: https://github.com/kohya-ss/musubi-tuner/blob/main/src/musubi_tuner/dataset/dataset_config.md#sample-for-image-dataset-with-control-images.

**Comment by icekeg** [comment]
The image dataset and the control dataset are one-to-one corresponding. I also use the jsonl file configuration method.

**Comment by icekeg** [comment]
The image dataset and the control dataset are one-to-one corresponding. I also use the jsonl file configuration method. 
The loss value during training is still nan. Could it be a problem with the image size of the dataset? The resolution of my image and control are both 540x1080 , here is the content of jsonl file and toml file. 

<img width="1820" height="212" alt="Image" src="https://github.com/user-attachments/assets/3de82c4e-4355-4426-a654-8c48ccafac4a" />

<img width="762" height="213" alt="Image" src="https://github.com/user-attachments/assets/dbeb1562-19d8-4683-8ecc-f824780903be" />

**Comment by kohya-ss** [comment]
It is normal for one cache file to be created for each pair of image and control image.

There are several factors that can cause the loss to become NaN. Could you please share your training settings (command line options)?

**Comment by icekeg** [comment]
Here are my training parameters:

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision fp16 src/musubi_tuner/flux_kontext_train_network.py \
    --dit /workspace/team_code_faidrawing/flux_kontext/flux1-kontext-dev.safetensors \
    --vae /workspace/team_code_faidrawing/flux_kontext/ae.safetensors \
    --text_encoder1 /workspace/team_code_faidrawing/flux_kontext/t5xxl_fp16.safetensors \
    --text_encoder2 /workspace/team_code_faidrawing/flux_kontext/clip_l.safetensors \
    --dataset_config /workspace/team_code_faidrawing/comfy_mtalk/train/kontext/blocksketch/sktch.toml \
    --sdpa \
    --mixed_precision fp16 \
    --timestep_sampling flux_shift \
	--weighting_scheme none \
    --optimizer_type adamw8bit \
	--learning_rate 1e-4 \
	--gradient_checkpointing \
    --max_data_loader_n_workers 2 \
	--persistent_data_loader_workers \
    --network_module networks.lora_flux \
	--network_dim 32 \
    --max_train_epochs 30 \
	--save_every_n_epochs 5 \
	--seed 42 \
    --output_dir /workspace/team_code_faidrawing/comfy_mtalk/train/kontext/blocksketch/sketch \
	--output_name blocksketch

**Comment by kohya-ss** [comment]
FLUX.1 Kontent has bf16 weights, not fp16, so please change `--mixed_precision` to `bf16`.

**Comment by icekeg** [comment]
Thank you very much，It works after I changed this setting.
